File: 0074-search-a-2d-matrix/0074-search-a-2d-matrix.py
class Solution:
    def searchMatrix(self, matrix: List[List[int]], target: int) -> bool:
        
        
        r, c = 0, len(matrix[0]) - 1
        
        
        while r >= 0 and c >= 0 and r < len(matrix) and c < len(matrix[0]):
            
            c_val = matrix[r][c]
            
            if target == c_val:
                return True
            if target < c_val:
                c -= 1
            else:
                r += 1
        

File: 0074-search-a-2d-matrix/0074-search-a-2d-matrix.cpp
class Solution {
public:
    bool searchMatrix(vector<vector<int>>& matrix, int target) {
        int row = 0;
        int col = matrix[0].size() - 1;

        while (col >= 0 && row < matrix.size()) {
            int cNum = matrix[row][col];
            if (target == cNum) return true;
            if (target > cNum) {
                row++;
            } else {
                col--;
            }
        }
        return false;
    }
};